aws-api/src/main/java/com/amplifyframework/api/aws/GsonListDeserializer.java

```diff
@@ -70,6 +70,6 @@
             }
         }
 
-        return new Gson().fromJson(json, typeOfT);
+        return GsonUtil.getGson().fromJson(json, typeOfT);
```

Comment:
Good callout!  Injecting via the constructor is a good idea, _but_ I actually realized that this class is not used at all, so I just deleted it :)

Reply:
If this class was used, it would actually be best to rely on the `JsonDeserializationContext` object that is already passed in.  You can't directly deserialize the JsonElement that is passed in since it results in endless recursion, but you could manually iterate through the array, and call `context.deserialize(arrayElement, clazz); on each one.`, and then return an array of all of those items.

I noticed there is a copy of GsonListDeserializer in DataStore (an inner class inside of AppSyncResponseDeserializer).    At some point, that needs to be refactored to use this approach instead of creating a new Gson() instance, because the new instance won't have all the type adapters (for dates/errors) needed to properly deserialize.